### PR TITLE
Adding wiki for Backup storage domain

### PR DESCRIPTION
**Pull request by shubham0d**
Adding wiki for Backup Storage Domain

Changes proposed in this pull request:
-
New wiki added which describe the documentation and functionality of Backup Storage.It is open for everyone to review and share their opinion.

Note:More detail and discussion can be found on ovirt user mailing list.

I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md): (@shubham0d)

### CODE REVIEW

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    <import>true</import>
+    <host id="9d05868b-d40d-4a8c-9a81-dbf09d654fba" />
+    <type>data</type>
+    <backup>true</backup>
+    <storage>
+        <type>fcp</type>
+    </storage>
+</storage_domain>
+```
+
```
Comment (maorlipchuk) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: Please remove redundant spaces

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+#### Advantages
+
+* Much more dedicated storage domain for backup and disaster recovery purposes.
+* One step process-If you want a backup of VM/Template then just transfer it to backup storage domain.
+* For huge amount of VMs/Templates and OVFs migration the time taken for copying through export will be high which will be minimized by using backup storage domain.
+* Large amount of extra space is used in export storage domain. It may be suitable for small datacenter but for large datacenter the storage required to store those OVFs and VMs in export storage domain will become a headache.
+* Multiple storage domain can be used for backup purposes only.
+* Backup storage domain will support both File storage(NFS, Gluster) as well as block storage(Fiber Channel and iSCSI) as compare to export storage domain which only has support for file storage.
+
+#### disadvantages
```
Comment (maorlipchuk) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: Probably better to change disadvantages to "restrictions"

Comment (maorlipchuk) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: probably better to rephrase this to restrictions instead of disadantages

**Comment by maorlipchuk** [comment]
Hi Shubham,
Looks very detailed, great work!

Let's wait for other reviewers to add their comments/opinions:
@bproffitt 
@mureinik 
@rollandf 
@mykaul 
@dyaniv

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
+---
+title: BackupStorageDomain
+category: feature
+authors: mlipchuk, shubham0d
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: @shubham0d you should add yourself to the `data/authors.yml` file too so your username renders nicely here.

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+wiki_category: Feature|BackupStorageDomain
+wiki_title: Features/BackupStorageDomain
+wiki_revision_count: 0
+wiki_last_updated: 2017-24-03
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: IIUC, all the `wiki*` fields are leftovers from the migration from the old wiki to the new site. I don't think we should use them in new pages.

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+feature_name: Backup Storage Domain
+feature_modules: engine/vdsm
+feature_status: Yet to Impelement
+--- 
+
+
+# Backup Storage Domain
+
+This feature will add another functionality layer to the oVirt Disaster Recovery solution.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: Let's beef this up a bit - what's the problem you're trying to solve with this feature? How about a sentence explaining that it's another step in deprecating the export domain?

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+
+## Current status
+
+* Yet to Implement
+
+
+## Summary
+
+Until now, one of the solutions to backup VMs and Templates in oVirt is to use an export storage domain.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: "is to use" -> "was to use"

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+Export storage domain is a dedicated storage domain which is used to store and restore VMs and Templates.
+The drawback of Export storage domain is its two stage process. Transferring a VM required first copying it into the export storage domain and then into other storage domain to start using them. Since in background transfer from one domain to other take place using normal copy or dd utility(in case of disks) this process take a large time for a complete transfer. Hence, for a better backup experience we decided to create a whole dedicated storage domain to be used only for backup which obviously will call as backup storage domain.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: missing space before "(in case of disks)"

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+
+## GSOC
+
+This feature is being developed as part of the Google Summer of Code (GSOC).
+Google Summer of Code is a global program focused on introducing students to open source software
+development. Students work on a 3 month programming project with an open source organization
+during their break from university.<br />
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: Can we add a link to GSOC's website?

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+The oVirt organization has chosen Shubham Dubey, a student from The LNM Institute of Information Technology from India, to work  on this project for the upcoming summer and Maor Lipchuk, a senior software developer in the oVirt storage team, to be the mentor.
+
+
+## Backup Storage Domain - Functionality
+
+* You can select any data storage domain as backup storage.
+* Once the storage domain is configured as backup the engine will block any running VMs or any changes that might be in the storage domain.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: "any changes" is ver wide term. We need better wording here. Something along the lines of "any action that may invalidate the backup"

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+* A backup storage domain can be detached and attached to a data center as every data storage domain.
+* Backup storage domain will be able to support unregistered VMs/Templates/Disks.
+* The backup indication will only be configured using the DataBase and will not be configured in the storage domain metadata - The user will be able to configure a storage domain as backup once a data storage domain is being imported or added to oVirt.
+* There is no obligation regarding data center version in oVirt to a backup storage domain. A backup storage domain can be part of every Data Center version.
+* A user can run a VM based on a Template which one of its disks is part of a backup storage domain while all the VM's disks are not part of the backup storage domain.
+
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: Isn't a "backup storage domain" just a data storage domain with an extra property? If so, a lot of the points here are redundant.

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+#### Advantages
+
+* Much more dedicated storage domain for backup and disaster recovery purposes.
+* One step process-If you want a backup of VM/Template then just transfer it to backup storage domain.
+* For huge amount of VMs/Templates and OVFs migration the time taken for copying through export will be high which will be minimized by using backup storage domain.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: The point that needs to be made here is that a backup domain can just be truned into a regular data domain at any instant.

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+* Large amount of extra space is used in export storage domain. It may be suitable for small datacenter but for large datacenter the storage required to store those OVFs and VMs in export storage domain will become a headache.
+* Multiple storage domain can be used for backup purposes only.
+* Backup storage domain will support both File storage(NFS, Gluster) as well as block storage(Fiber Channel and iSCSI) as compare to export storage domain which only has support for file storage.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: missing spaces before the commas

"as compare" -> "as compared"

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+
+#### Restrictions
+
+* A data storage domain can not be configured as backup while there are running VMs with disks reside on that storage domain.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: "running" -> "not down".
we want to also block paused, hibernated, powering up/down, etc, VMs.

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+* User can not run VMs with disks reside on a storage domain configured as backup, since running VM might manipulate the disk's data.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: reside -> residing

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+* VMs with disks reside on a backup storage domain can not be previewed.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: here too

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+* Live move of disks to the backup storage domain will be restricted.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: Additionally: 

1. A backup domain cannot be elected as the master domain
2. The Hosted Engine's domain can't be configured as a backup domain (although this may be redundant as it always has the hosted engine VM which is up, but better be safe than sorry
3. The backup domain cannot be the target of memory volumes
4. The backup domain cannot hold VM leases (need to think about this - perhaps we should allow creating them but not running VMs with them?)

Open question - can you edit a VM/disk that's on the backup domain?

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+
+#### Open Issues
+
+* Preview will be restricted for VMs with disks reside on a backup storage. - We think it should be restricted since currently oVirt does not support import storage domain with previewed unregistered entities.
+* Shared disk will be restricted in the backup storage domain since those are not specified in the VM's OVF.
+* VM pool will be eligible in a backup storage domain although, the user must keep in mind that import storage domain will not preserve its pool reference after import.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: This is a bug (RFE?) that needs to be addressed regardless of the backup domain

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 18]
```diff
@@ -0,0 +1,248 @@
...
+* Should the backup indication needs to be configured in the storage domain meta data? If so should we add the backup indication as part of V4 storage domain meta data.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: needs -> need

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 19]
```diff
@@ -0,0 +1,248 @@
...
+  * introducing new field 'backup' in table storage_domain_static
+  * Add field changes in fixtures.xml for dao tests
+  * Test class added for dao test
+- [ ] Phase 2: Add command validations for configuring backup storage domain - [see restrictions]
+- [ ] Phase 3: Add REST command to update storage domain as backup - Introduce the ability to update the storage domain as backup through REST.
+
+
+## Future plans
+
+* Add GUI support for backup storage domain - Once backend configuration get complete and REST changes get implemented we will make this ready for user to use it through gui.
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: gui -> GUI

File: source/develop/release-management/features/storage/backup-storage-domain.html.md [hunk 20]
```diff
@@ -0,0 +1,248 @@
...
+ An Import storage domain (NFS, iSCSI) screen:
+![](/images/wiki/Import_with_backup_NFS.png)
+
+ An Add New Storage Domain (NFS, iSCSI) screen:
+![](/images/wiki/Add_new_domain_with_backup_NFS.png)
+
+
+## REST Requests
+
+### Import block Storage Domain
```
Comment (mureinik) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: What do any of these have to do with this feature?

Comment (shubham0d) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: if you are talking about GUI mockups and REST requests then they are basically for demonstration purpose so reader will know how this feature can be used. But I can remove them if they are not necessary.

File: source/develop/release-management/features/storage/backup-storage-domain.html.md (hunk 16 above)
Comment (shubham0d) [inline source/develop/release-management/features/storage/backup-storage-domain.html.md]: @mureinik we are not allowing running VMs which have disks on
backup storage domains so the user will not be able to create memory
volumes since those are only supported for running VMs.

**Comment by mykaul** [comment]
ping?

**Comment by shubham0d** [comment]
I think this is ready to submit.

**Comment by sandrobonazzola** [comment]
@shubham0d can you please rebase this or close the PR?
